Modules/_zoneinfo.c

```diff
@@ -224,8 +224,14 @@ zoneinfo_new_instance(PyTypeObject *type, PyObject *key)
     self = NULL;
 cleanup:
     if (file_obj != NULL) {
+        PyObject *exc, *val, *tb;
+        PyErr_Fetch(&exc, &val, &tb);
         PyObject *tmp = PyObject_CallMethod(file_obj, "close", NULL);
-        Py_DECREF(tmp);
+        _PyErr_ChainExceptions(exc, val, tb);
+        Py_XDECREF(tmp);
+        if (tmp == NULL) {
```

Comment:
It is an undefined behavior to compare a pointer to the freed memory with NULL. Do it before `Py_XDECREF(tmp)`.